collect_app/src/main/java/org/odk/collect/android/preferences/LocaleHelper.java

```diff
@@ -0,0 +1,60 @@
+package org.odk.collect.android.preferences;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.util.DisplayMetrics;
+
+import java.util.Locale;
+import java.util.TreeMap;
+
+/**
+ * Changes the locale of the app and keeps the changes persistent
+ *
+ * @author abdulwd
+ */
+
+public class LocaleHelper {
```

Comment:
The nice thing about keeping it as an instantiable class is that we can then put it under unit test. getEntryListValues and getLocale should be under test.